Lib/concurrent/futures/_base.py

```diff
@@ -170,6 +170,17 @@ def _create_and_install_waiters(fs, return_when):
 
     return waiter
 
+
+def _yield_and_decref(fs, waiter, ref_collect=()):
```

Comment:
This is my best, i could not came up with anything better. Maybe you can suggest name more descriptive.